TypeCobol/Compiler/Types/Type.cs

```diff
@@ -0,0 +1,282 @@
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The Symbol associated to this type if any: This for a Program or a Function or a TYPEDEF
+        /// </summary>
+        public Symbol Symbol
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The length of this type.
+        /// </summary>
+        public virtual int Length
+        {
+            get
+            {
+                switch (Usage)
+                {
+                    case UsageFormat.Comp:
+                    case UsageFormat.Comp4:
+                    case UsageFormat.Comp5:
+                    case UsageFormat.Display1:
+                    case UsageFormat.National:
+                    case UsageFormat.Binary:
+                        return 2;
+                    //Floating-point: Specifies for internal floating -point items (single precision)
+                    //(i.e float in java, or C)
+                    case UsageFormat.Comp1:
+                    case UsageFormat.FunctionPointer:
+                    case UsageFormat.ObjectReference:
+                    case UsageFormat.Index:
+                    case UsageFormat.Pointer:
+                        return 4;
+                    //Long floating-point: Specifies for internal  floating point items(double precision)
+                    //(i.e double in java or C)
+                    case UsageFormat.Comp2:
+                    case UsageFormat.ProcedurePointer:
+                        return 8;
+                    case UsageFormat.Comp3:
+                    case UsageFormat.Display:
+                    case UsageFormat.PackedDecimal:
+                        return 1;
+
+                    default:
+                        throw new ArgumentException("Invalid Usage for type length calculation : " + Usage.ToString());
+                }
+            }
+        }
+
+        public SemanticKinds SemanticKind => SemanticKinds.Type;
+
+        /// <summary>
+        /// Set a set of flags to true or false.
+        /// </summary>
+        /// <param name="flag"></param>
+        /// <param name="value"></param>
+        internal virtual void SetFlag(Flags flag, bool value)
+        {
+            this.Flag = value ? (Flags)((ulong)this.Flag | (ulong)flag)
+                : (Flags)((ulong)this.Flag & ~(ulong)flag);
+        }
+
+        /// <summary>
+        /// Determines if the given flag is set.
+        /// </summary>
+        /// <param name="flag">The flag to be tested</param>
+        /// <returns>true if yes, false otherwise.</returns>
+        public bool HasFlag(Flags flag)
+        {
+            return ((ulong)this.Flag & (ulong)flag) != 0;
+        }
+
+        public object Clone()
+        {
+            return MemberwiseClone();
+        }
+
+        /// <summary>
+        /// TypeComponent for example for Array, Pointer type or TypeDef.
+        /// </summary>
+        public virtual Type TypeComponent => null;
```

Comment:
The comment doesn't explain what is the purpose of TypeComponent. Can you explain in the comment ?

Reply:
ok